packages/sample-controllers/src/sample-petnames-controller.test.ts

```diff
@@ -19,160 +22,237 @@ describe('SamplePetnamesController', () => {
           },
         },
       };
-      const controller = new SamplePetnamesController({
-        messenger: getMessenger(),
-        state: givenState,
-      });
 
-      expect(controller.state).toStrictEqual(givenState);
+      await withController({ state: givenState }, ({ controller }) => {
+        expect(controller.state).toStrictEqual(givenState);
+      });
     });
 
-    it('fills in missing state properties with default values', () => {
-      const controller = new SamplePetnamesController({
-        messenger: getMessenger(),
+    it('fills in missing state properties with default values', async () => {
+      await withController(({ controller }) => {
+        expect(controller.state).toMatchInlineSnapshot(`
+          Object {
+            "namesByChainIdAndAddress": Object {},
+          }
+        `);
       });
-
-      expect(controller.state).toMatchInlineSnapshot(`
-        Object {
-          "namesByChainIdAndAddress": Object {},
-        }
-      `);
     });
   });
 
-  describe('assignPetname', () => {
+  describe.each([
```

Comment:
This seems a bit convoluted 🤔 Is it helpful to test the method in both ways? If we test just the method, certainly the tests would miss a problem with the action registration. But if we have tests for the action, it's unclear what value the tests for the method have. Especially if we decide to disallow calling methods directly at some point in the future.

Reply:
I have noticed that it is easy when adding an action which directly maps to a method to forget to test the action. This approach seemed like the best way to ensure that both the method and the action do exactly the same thing.

But if you think this is convoluted, and you think that at some point we will disallow calling methods at some point in the future, perhaps we should promote a pattern where the action is tested comprehensively and the method is tested more simply?

Reply:
Yeah, I know what you mean. We could certainly ask people to stop testing methods directly that are being exposed via the messenger. I worry that this guidance would be out of place at the moment, though. As it is, we _do_ use methods directly, both in the background API and in Mobile. So at the moment, it is useful, even if it's not ideal.